### PR TITLE
feat(filters): Add base64_encode and base64_decode filters for Shopify compatibility

### PR DESCRIPTION
**Pull request by rosomri**
### Summary
This PR adds two standard Shopify filters (`base64_encode` and `base64_decode`) to LiquidJS core to improve Shopify compatibility. These filters are part of the standard Shopify Liquid filter set but were not previously implemented in LiquidJS.

https://github.com/harttle/liquidjs/issues/827

### Changes
- **New Filters**: Added `base64_encode` and `base64_decode` filters with full Shopify compatibility
- **Platform-specific Implementation**: 
  - Node.js: Uses `Buffer.from()` for optimal performance
  - Browser: Uses `btoa`/`atob` with proper Unicode handling
- **Rollup Integration**: Added bundle options to replace implementations during build
- **Comprehensive Testing**: Added tests for both Node.js and browser implementations
- **Memory Management**: Integrated with LiquidJS's memory limit system

### Implementation Details
- `src/filters/base64.ts` - Main filter implementation
- `src/filters/base64-impl.ts` - Node.js implementation using Buffer
- `src/build/base64-impl-browser.ts` - Browser implementation using btoa/atob
- `test/integration/filters/base64.spec.ts` - Integration tests
- `src/build/base64-impl-browser.spec.ts` - Browser-specific unit tests
- Updated `rollup.config.mjs` with platform-specific bundling

### Shopify Compatibility
- ✅ Matches Shopify's exact output for `{{ 'one two three' | base64_encode }}` → `b25lIHR3byB0aHJlZQ==`
- ✅ Handles Unicode characters correctly
- ✅ Supports all input types (strings, numbers, booleans)
- ✅ Proper error handling for edge cases

### Testing
- unit tests for browser implementation
- integration tests covering all scenarios
- Round-trip encoding/decoding verification
- Unicode and special character handling
- Cross-platform compatibility verified

### Breaking Changes
None

### CODE REVIEW

File: src/build/base64-impl-browser.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+
+export function base64Encode (str: string): string {
+  return btoa(String.fromCharCode(...new TextEncoder().encode(str)))
```
Comment (harttle) [inline src/build/base64-impl-browser.ts]: Try avoid TextEncoder and Uint8Array as we don’t heavily depend on them. Use escape and decodeURIComponent trick for better compatibility.

**Review by harttle — CHANGES_REQUESTED** [review]
